kolibri/core/assets/src/views/CoreMenu/CoreMenuOption.vue

```diff
@@ -83,12 +130,52 @@
       optionIconColor() {
         return this.active ? this.$themeTokens.primary : null;
       },
+      iconAfter() {
+        return this.visibleSubMenu ? 'chevronUp' : 'chevronDown';
+      },
+    },
+    mounted() {
+      this.submenuShouldBeOpen();
     },
     methods: {
-      conditionalEmit() {
-        if (!this.link) {
-          this.$emit('select');
+      isActiveLink(route) {
+        return `${this.link}#${route}` === `${window.location.pathname}${window.location.hash}`;
+      },
+      submenuShouldBeOpen() {
+        if (this.subRoutes && this.subRoutes.length > 0) {
+          window.location.pathname === this.link
+            ? (this.visibleSubMenu = true)
+            : (this.visibleSubMenu = false);
         }
+        return false;
+      },
+      subpathStyles(route) {
+        if (this.isActiveLink(route)) {
+          return {
+            color: this.$themeTokens.primaryDark,
+            fontWeight: 'bold',
+            textDecoration: 'none',
+          };
+        }
+        return {
+          color: this.$themeTokens.text,
+          textDecoration: 'none',
+          ':hover': {
+            color: this.$themeTokens.primaryDark,
+            fontWeight: 'bold',
+          },
+          ':focus': this.$coreOutline,
+        };
+      },
+      toggleAndroidMenu() {
+        this.$emit('toggleAndroidMenu');
+      },
+      handleNav(route) {
+        this.isActiveLink(route) ? this.toggleAndroidMenu() : null;
+      },
+      generateNavRoute(route) {
+        const params = this.$route.params;
+        return generateNavRoute(this.link, route, params);
```

Comment:
Feels like this could still be conditionalized to use vue router in the case we are on the SPA?